src/D2L.CodeStyle.Analyzers/Immutability/MutabilityAuditor.cs

```diff
@@ -3,15 +3,24 @@
 
 namespace D2L.CodeStyle.Analyzers.Immutability {
 	internal sealed class MutabilityAuditor {
-		public static bool IsAudited( ISymbol symbol, out Location location ) {
+		public static bool IsAudited( ISymbol symbol, out Location location, out Diagnostic diagnostic ) {
 			AttributeData attr;
+			diagnostic = null;

```

Comment:
Early assignment of out parameters is an anti-pattern. This is usually done to defeat CS0177, however the compiler error is also there to help you. Assigning at the top means it won't catch if you forgot to assign a proper value in a given case.